django/website/logframe/views.py

```diff
@@ -58,3 +62,31 @@ def get_data(self, logframe, data):
             'result': ResultSerializer(self.object).data,
         })
         return data
+
+
+class CreateLogframe(PermissionRequiredMixin, CreateView):
+    model = LogFrame
+    fields = ['name']
+    template_name = 'logframe/create_logframe.html'
+    permission_required = 'logframe.edit_logframe'
+
+    def get_unique_slug_name(self, logframe):
+        slug = logframe.name.lower()
+        slug = slug.replace(' ', '_')
+        slug = re.sub('[^\w-]', '', slug)
+        slug = slug[:46]
+        if LogFrame.objects.filter(slug=slug).exists():
+            count = LogFrame.objects.filter(slug__startswith=slug).count() + 1
+            slug += unicode(count)
```

Comment:
It wouldn't avoid duplicate code in the migration, because migrations use their own model classes, not the ones in the models files. However, I have moved the code to the models file and made it so that the code is incremented until the model is unique.